docs/miscellaneous.rst

```diff
@@ -58,6 +61,51 @@ So for the following contract snippet::
 
 The position of ``data[4][9].b`` is at ``keccak256(uint256(9) . keccak256(uint256(4) . uint256(1))) + 1``.
 
+Adding elements to or removing them from a dynamically-sized array will increment or decrement the number
+of elements which is stored in the slot. For arrays that are not of type ``bytes`` or ``string`` this is
+straight-forward, since data is stored in a different slot.
+
+
+``bytes`` and ``string``
+------------------------
+
+``bytes`` and ``string`` store their data in the same slot where also the length is stored if they are short. In
+particular: If the data is at most ``31`` bytes long, it is stored in the higher-order bytes (left aligned) and the
+lowest-order byte stores ``length * 2``. If it is longer, the main slot stores ``length * 2 + 1`` and the data is
+stored as usual in ``keccak256(slot)``.
+
```

Comment:
Perhaps explain that the least significant bit can be used to determine how to decode the main slot, i.e. whether it only stores the length or also data.